// asynchronous-local-storage/dist/lib/nodeVersionUtils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isAlsSupported = exports.getNodeVersion = exports.nodeVersionString = void 0;
exports.nodeVersionString = process.versions.node;
function getNodeVersion(nodeVersionString) {
    const [nodeMajor, nodeMinor, nodePatch] = nodeVersionString.split('.');
    return {
        majorVersion: Number.parseInt(nodeMajor),
        minorVersion: Number.parseInt(nodeMinor),
        patchVersion: Number.parseInt(nodePatch),
    };
}
exports.getNodeVersion = getNodeVersion;
function isAlsSupported(nodeVersion) {
    const { majorVersion, minorVersion } = nodeVersion;
    if (majorVersion > 13) {
        return true;
    }
    if (majorVersion < 12) {
        return false;
    }
    // https://nodejs.org/en/blog/release/v12.17.0/
    if (majorVersion === 12) {
        return minorVersion >= 17;
    }
    // https://nodejs.org/en/blog/release/v13.10.0/
    if (majorVersion === 13) {
        return minorVersion >= 10;
    }
    return false;
}
exports.isAlsSupported = isAlsSupported;
